### DISCUSSION TITLE
Should I Open an Issue or Raise a PR First?

### DISCUSSION THREAD

**Discussion by rishabh3562** [community/community#140951] [Pull Requests] [answered]
### Select Topic Area

Question

## Should I Open an Issue or Raise a PR First?

Hey everyone!

I’m new to the open source world and have been trying to contribute to some projects. I have a question about best practices: 

Is it better to raise a PR directly or open an issue first?

I’ve been making some random contributions, but they aren’t tied to specific issues. Should I just go ahead and raise a PR, or is there a standard practice I should follow? The contributing guidelines don’t clarify much.

Could you also provide some steps, like whether I should create a fork, how to name my branch, and how to format my commits?

Any advice would be appreciated! Thanks!

**Comment by mehulmina064** [comment, marked as answer]
- If you plan on adding any new features or making changes to something that has not been thoroughly discussed, please open an issue first. It helps the maintainers know what you are working on.

- If it’s a minor adjustment (such as a typo or small bug) that doesn’t need much discussion or explanation, just raise a PR directly.

**Comment by AudreyLin** [comment]
Hi @rishabh3562.  You requested more information about contributing to open source.  This  [Opensource.guide](https://opensource.guide/how-to-contribute/) covers all you need to know about contributing to open-source.  Also, here is a guide on [How to Make A Pull Request](https://makeapullrequest.com/).  It sounds like you've already made your first contributions, but here is a repo with the steps to make your [First Contributions](https://github.com/firstcontributions/first-contributions).  These resources should provide more answers to your questions.  Happy Hacking.
